### DISCUSSION TITLE
Add friendly errors for foreign key constraint problems

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#17079] [Feature Requests]
When deleting an image, that is prevented from deleting, via the `File Library`, we get an “Unexpected Error” that is not human readable. Would be nice to get a message like “File is used by XXX and not allowed to delete”.

### To Reproduce

Make a collection with a single file field and set the deletion option to `Prevent deleting`, then go to the `File Library` and try to delete.

### Errors Shown

```
{
  "errors": [
    {
      "message": "delete from `directus_files` where `id` in ('...') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`...`.`...`, CONSTRAINT `..._foreign` FOREIGN KEY (`...`) REFERENCES `directus_files` (`id`) ON DELETE NO ACTION)",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}f
```

**Comment by azrikahar** [comment]
This improvement should fall under the same scope in #9507 👍

**Comment by azrikahar** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
